### PR TITLE
feat: move clipboard module into cdk

### PR DESCRIPTION
**Pull request by crisbeto**
Moves the `clipboard` module into CDK stable, sets up the API goldens, adjusts some APIs to be more consistent and sets up a live example.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -66,6 +66,7 @@
 /src/cdk/a11y/**                                   @jelbourn @devversion
 /src/cdk/accordion/**                              @jelbourn
 /src/cdk/bidi/**                                   @jelbourn
+/src/cdk/clipboard/**                              @jelbourn
```
Comment (crisbeto) [inline .github/CODEOWNERS]: Not sure whether we want any other people to own this.

File: src/cdk/clipboard/copy-to-clipboard.ts [hunk 2]
```diff
@@ -21,18 +21,23 @@ import {Clipboard} from './clipboard';
...
+   */
   @Output() copied = new EventEmitter<boolean>();
 
-  constructor(private readonly clipboard: Clipboard) {}
+  constructor(private readonly _clipboard: Clipboard) {}
 
-  doCopy() {
-    this.copied.emit(this.clipboard.copy(this.text));
+  /** Copies the current text to the clipboard. */
+  copy() {
```
Comment (crisbeto) [inline src/cdk/clipboard/copy-to-clipboard.ts]: I renamed this to align it with the service and the `PendingCopy`, but I'm not totally sure whether it should be public in the first place.

File: .github/CODEOWNERS (hunk 1 above)
Comment (jelbourn) [reply .github/CODEOWNERS]: You can add `@xkxx`

File: src/cdk/clipboard/clipboard.md [hunk 3]
```diff
@@ -1,8 +1,17 @@
...
+
+The `cdkCopyToClipboard` directive can be used to easily add copy-on-click
+functionality to an existing element. The directive selector doubles as an
+`@Input()` for the text to be copied.
+
+```html
+<img src="avatar.jpg" alt="Hero avatar" [cdkCopyToClipboard]="getShortBio()">
+```
+
+<!-- example(cdk-platform-overview) -->
```
Comment (jelbourn) [inline src/cdk/clipboard/clipboard.md]:
```html
<!-- example(cdk-clipboard-overview) -->
```

File: src/cdk/clipboard/copy-to-clipboard.ts (hunk 2 above)
Comment (jelbourn) [reply src/cdk/clipboard/copy-to-clipboard.ts]: I think it's fine being public

File: src/material-examples/cdk/clipboard/cdk-clipboard-overview/cdk-clipboard-overview-example.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+ */
+@Component({
+  selector: 'cdk-clipboard-overview-example',
+  templateUrl: 'cdk-clipboard-overview-example.html',
+  styleUrls: ['cdk-clipboard-overview-example.css'],
+})
+export class CdkClipboardOverviewExample {
+  value = 'Lorem, ipsum dolor sit amet consectetur adipisicing elit. Ex fugiat recusandae ' +
+          'consectetur illoet alias. Quia, numquam alias. Modi dolorum hic perferendis ' +
+          'reiciendis doloribus officiis aliquid asperiores repellat deleniti rerum.';
```
Comment (jelbourn) [inline src/material-examples/cdk/clipboard/cdk-clipboard-overview/cdk-clipboard-overview-example.ts]: optional: I always like using text like this as an opportunity as something fun. Like, maybe part of the theme song to _Fresh Prince of Bel Air_

**Comment by crisbeto** [comment]
I've addressed the feedback @jelbourn.

File: src/material-examples/cdk/clipboard/cdk-clipboard-overview/cdk-clipboard-overview-example.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+/**
+ * @title Clipboard overview
+ */
+@Component({
+  selector: 'cdk-clipboard-overview-example',
+  templateUrl: 'cdk-clipboard-overview-example.html',
+  styleUrls: ['cdk-clipboard-overview-example.css'],
+})
+export class CdkClipboardOverviewExample {
+  value = `Did you ever hear the tragedy of Darth Plagueis The Wise? I thought not. It's not ` +
```
Comment (jelbourn) [inline src/material-examples/cdk/clipboard/cdk-clipboard-overview/cdk-clipboard-overview-example.ts]: A surprise for sure, but a welcome one.

**Comment by jelbourn** [comment]
Caretaker note: this will require updating build rules internally

**Pull request merged**